### PR TITLE
[ROCm] Enable bfloat16 for ops in BERT model

### PR DESCRIPTION
**Pull request by rohithkrn**
Enables bfloat16 type for ops present in BERT model.
Enabled relevant unit tests.

@ezyang @jeffdaily

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -17267,18 +17267,18 @@ def inner(self, device, dtype):
         1e-5, 1e-5, 1e-5, _float_types_no_half, _cpu_types, False),
     ('addcdiv', '', _small_2d,
         lambda t, d: [_small_2d(t, d),
-                      _small_2d(t, d, has_zeros=False)], 1, 1e-5, 1e-3,
-        _types, _cpu_types, True,
```
Comment (ezyang) [inline test/test_torch.py:17271]: @nairbv could you please review the precision changes?

**Comment by ezyang** [comment]
Code changes look fine. @nairbv has been working on making precision in our test suite more uniform, and I'd like it if he could take a look at the changes here.

File: test/test_torch.py (hunk 1 above)
Comment (nairbv) [reply test/test_torch.py:17271]: prec changes look fine.

**Review by nairbv — APPROVED** [review]
I'm just validating the prec changes, but I see @ezyang says "Code changes look fine" so marking the PR approved.

**Pull request closed without merging**